plugins/auth-backend/src/identity/IdentityClient.ts

```diff
@@ -28,16 +34,37 @@ const CLOCK_MARGIN_S = 10;
  * @experimental This is not a stable API yet
  */
 export class IdentityClient {
-  private readonly discovery: PluginEndpointDiscovery;
-  private readonly issuer: string;
-  private keyStore: JWKS.KeyStore;
-  private keyStoreUpdated: number;
+  static async create(options: {
+    database: PluginDatabaseManager;
+    discovery: PluginEndpointDiscovery;
+    logger: Logger;
+  }): Promise<IdentityClient> {
+    const issuer = await options.discovery.getExternalBaseUrl('auth');
+    return new IdentityClient(
+      issuer,
+      new TokenFactory({
+        issuer: issuer,
+        logger: options.logger.child({ component: 'token-factory' }),
+        keyStore: await DatabaseKeyStore.create({
```

Comment:
I'm extremely hesitant towards this to be honest. It's not good to have a collection of different microservices using the same database, which is what we end up with here. What do you think about exploring a more direct shared secret approach to service-to-service auth that operates separately from the OpenID-like approach that we use for the user tokens?